app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.add_alive()
+
+    def add_alive(self) -> None:
+        if self.health > 0 and self not in Animal.alive:
```

Comment:
You are accessing the class attribute `alive` via an instance `self`. It is better to use the class name `Animal` to get access to the class attributes. This is important to avoid confusion between class and instance attributes. Please refer to the checklist, Code Efficiency point 1.